### ISSUE TITLE
hunspell only supports 2 suffixes (and/or prefixes) per word

### ISSUE THREAD

**Issue by mhosken** [hunspell/hunspell#497]
Although the hunspell .aff file supports unlimited suffix/prefix chaining, only 2 levels are supported in the engine. Is there a reason for this limitation. There are many languages out there that use many more affix levels than 2 (even English - interlinearisations is 3 to start with). What were the 'we can't let it go in indefinitely because' issues that were encountered when going to 2?

**Comment by shantanuo** [comment]
@mhosken this is a show-stopper. The limit of nesting should be increased to atleast 3 levels.

# cat en_US2.aff
SFX     Q       Y       1
SFX     Q       0       ed/e      .

SFX     e       Y       1
SFX     e       0       s/x      .

SFX    x       Y       1
SFX     x       0       fe      .


# cat en_US2.dic
1
test/Q

It will return the following words but the word ‘testedsfe’ is not listed.

test
tested
testeds

Most of the Indian Languages like Sanskrit, Marathi need this.

**Comment by adamb924** [comment]
Yes, there is a large number of languages for which this is a problem.

The workaround that people in aggulatinative languages seem to use is generating a long list of possible affixes—I hope from an abstract morphological model!
